src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -47,11 +47,32 @@ public function has($relation, $operator = '>=', $count = 1, $boolean = 'and', C
             $hasQuery->callScope($callback);
         }
 
+        // When query connection and subquery connection are not equals concat connection to query from
+        if ($this->hasCanCrossDatabase($hasQuery)) {
+            $subqueryConnection = $hasQuery->getConnection()->getDatabaseName();
+            $queryConnection = $this->getConnection()->getDatabaseName();
+            if ($queryConnection != $subqueryConnection) {
+                $queryFrom = $hasQuery->getQuery()->from.'|'.$subqueryConnection;
+                $hasQuery->from($queryFrom);
+            }
+        }
+
         return $this->addHasWhere(
             $hasQuery, $relation, $operator, $count, $boolean
         );
     }
 
+    protected function hasCanCrossDatabase($hasQuery)
+    {
+        foreach (['MySqlConnection', 'PostgresConnection', 'SqlServerConnection'] as $connector) {
+            if (is_a($this->getConnection(), '\Illuminate\Database\\'.$connector) && is_a($hasQuery->getConnection(), '\Illuminate\Database\\'.$connector)) {
+                return true;
```

Comment:
This looks problematic; what if I have a custom database connection? Would it be possible to check for an interface instead? `return $this->getConnection() instanceof CanCrossDatabaseShazaamInterface;`

Reply:
I like sisve's idea more